tree-walk/src/interpreter.h

```diff
@@ -33,12 +34,16 @@ struct InterpreterVisitor {
   Value operator()(const ast::Assign &assign);
   Value operator()(const ast::Binary &bin);
   Value operator()(const ast::Call &call);
+  Value operator()(const ast::Get &get);
   Value operator()(const ast::Grouping &grp);
   Value operator()(const ast::Literal &ltrl);
+  Value operator()(const ast::Set &set);
   Value operator()(const ast::Unary &unary);
   Value operator()(const ast::Variable &var);
 
 private:
+  Value invoke(const FnClosureShrdPtr &fnSPtr, const ast::Call &call);
+  Value invoke(const ClsFactShrdPtr &factSPtr, const ast::Call &call);
```

Comment:
Yeah this might be nicer if we store the call in the cases we want to execute those. Or we create a new visitor for that? Might seem a bit weird having multiple visitors for the interpreter but actually we already have that anyway with the AdditionVisitor etc. Wouldn't make sense to initialise this visitor with a Call object most of the time so maybe it should be a new object.